src/rpc/rpc_impl.rs

```diff
@@ -337,4 +341,26 @@ impl<N: Network, E: Environment> RpcFunctions<N> for RpcImpl<N, E> {
         }
         Ok(transaction.transaction_id())
     }
+
+    /// Returns the amount of shares submitted by a given prover.
+    async fn get_shares_for_prover(&self, prover: Value) -> Result<u64, RpcError> {
+        let prover: Address<N> = serde_json::from_value(prover)?;
+        Ok(self.operator.get_shares_for_prover(&prover))
+    }
+
+    /// Returns the amount of shares submitted to an operator in total.
+    async fn get_shares(&self) -> Result<u64, RpcError> {
```

Comment:
```suggestion
    async fn get_shares(&self) -> u64 {
```
there is no room for error here, we should avoid it if possible